src/client/components/ModalNew.tsx

```diff
@@ -7,6 +7,8 @@ import { Cross } from './icons/Cross'
 export interface ModalProps {
   isVisible: boolean
   dynamicHeight?: boolean
+  maxWidth?: string
+  overflowX?: string
```

Comment:
I would prefer passing `className` to the modal and styling with:

```
const MyCustomModal = styled(Modal)`
  maxWidth: 23rem;
`
```

That is unless these values are meant to be truly dynamic :)

Reply:
Yeah this seems a bit hacky I agree, but correct me if I'm wrong: Wouldn't your suggestion just add "maxWidth" to the root-element of `Modal`, which is the `<Wrapper>` and not the actual modal dialog?

Reply:
Imo this is the real culprit
```const ModalInnerContainer = styled('div')`
  width: 100%;
  height: 100%;
`
```

This essentially makes previous max-width a fixed width. Shouldn't we to some extent let the content decide the size of the modal?

Reply:
Aha, yeah that seems a little unnecessary I agree :/

You could technically pass a `className` prop to any of the child components of `<Modal>` -- whichever one makes most sense to style. And we we need more flexibility I would export multiple styled components from the module.

Reply:
Ah cool! Wasn't aware of this :)